code/game/machinery/vending.dm

```diff
@@ -95,17 +92,12 @@
 		build_inventory(products, product_records)
 		build_inventory(contraband, hidden_records)
 		build_inventory(premium, coin_records)
-	if(product_slogans)
-		slogan_list += splittext(product_slogans, ";")
-
+	if(slogan_list.len > 0)
```

Comment:
`if(LAZYLEN(slogan_list)`

There's another `slogan_list.len > 0` that needs to be corrected, somewhere.

Reply:
Changed.